src/lib/compat.h

```diff
@@ -31,13 +31,15 @@ int rand_r(unsigned int*) __attribute__((deprecated("Do not use rand_r, use i_ra
 /* NOLINTEND(readability-redundant-declaration) */
 #endif
 
-#if defined (HAVE_UOFF_T)
+#if defined(__APPLE__)
+typedef size_t uoff_t;
```

Comment:
This is a workaround for our broken code. The real fix is:

```
diff --git a/src/lib-http/http-server.h b/src/lib-http/http-server.h
index 91412b570b..c27170777a 100644
--- a/src/lib-http/http-server.h
+++ b/src/lib-http/http-server.h
@@ -49,8 +49,8 @@ struct http_server_settings {
        /* The kernel send/receive buffer sizes used for the connection sockets.
           Configuring this is mainly useful for the test suite. The kernel
           defaults are used when these settings are 0. */
-       size_t socket_send_buffer_size;
-       size_t socket_recv_buffer_size;
+       uoff_t socket_send_buffer_size;
+       uoff_t socket_recv_buffer_size;
 };
```

Do you see such errors elsewhere?

Reply:
pigeonhole-2.4.0 in `src/lib-sieve/plugins/variables/ext-variables-settings.c` ?

Reply:
Ah, yes, and `src/lib-sieve/sieve-settings.h` also. Fixing these.